private/buf/bufctl/controller.go

```diff
@@ -54,10 +54,13 @@ import (
 	"google.golang.org/protobuf/proto"
 )
 
-// ImageWithConfig pairs an Image with lint and breaking configuration.
+// ImageWithConfig pairs an Image with its corresponding [bufmodule.Module] full name
+// (which may be nil), [bufmodule.Module] opaque ID, and lint and breaking configurations.
 type ImageWithConfig interface {
 	bufimage.Image
 
+	ModuleFullName() bufparse.FullName
+	ModuleOpaqueID() string
```

Comment:
Maybe this could be `Module() bufmodule.Module` and document that it may be nil? From the docs it seems like the `Module.FullName` might be empty (like in Module) not that it isn't a Module,  as the opaque ID can be empty.

Reply:
Discussed outside of the PR: this is mostly for the case of `messageRef`, where we resolve the message through the bucket without the `Module`, so there is no `FullName` or `OpaqueID`. In those cases, we are not able to use `--against-registry` flag.